x-pack/plugins/features/server/feature_registry.ts

```diff
@@ -41,6 +42,20 @@ export class FeatureRegistry {
       throw new Error(`Feature with id ${feature.id} is already registered.`);
     }
 
+    if (!feature.scope) {
+      feature.scope = [KibanaFeatureScope.Security];
+    }
+
+    if (
+      feature.scope.length === 1 &&
+      feature.scope[0] === KibanaFeatureScope.Spaces &&
+      feature.privileges
+    ) {
+      throw new Error(
+        `Feature with id ${feature.id} and only spaces scope could not have privileges.`
+      );
+    }
```

Comment:
This can alternatively be enforced within `feature_schema` by using `schema.conditional`, similar to what we did for sub-feature privileges:

https://github.com/elastic/kibana/blob/88354146e4501ff305edac005e2520e274b95bee/x-pack/plugins/features/server/feature_schema.ts#L234-L240

Reply:
yeah, you are right, looks better!

Reply:
Added in 1d317e70ad6d382722e5bd0d5aea784ddca39ec2